chrome-launcher/chrome-launcher.ts

```diff
@@ -19,46 +19,77 @@
 
 import * as childProcess from 'child_process';
 import * as fs from 'fs';
-import * as path from 'path';
 import * as chromeFinder from './chrome-finder';
-import {ask} from './ask';
+import {getRandomPort} from './random-port';
 import {DEFAULT_FLAGS} from './flags';
-
-const mkdirp = require('mkdirp');
+import {makeTmpDir, defaults, delay} from './utils';
 import * as net from 'net';
 const rimraf = require('rimraf');
 const log = require('../lighthouse-core/lib/log');
 const spawn = childProcess.spawn;
 const execSync = childProcess.execSync;
 const isWindows = process.platform === 'win32';
+const _SIGINT = 'SIGINT';
+const _SIGINT_EXIT_CODE = 130;
+const _SUPPORTED_PLATFORMS = new Set(['darwin', 'linux', 'win32']);
+
+type SupportedPlatforms = 'darwin'|'linux'|'win32';
+
+export interface Options {
+  startingUrl?: string;
+  chromeFlags?: Array<string>;
+  port?: number;
+  handleSIGINT?: boolean;
+  chromePath?: string;
+}
 
-export class ChromeLauncher {
-  prepared = false;
-  pollInterval: number = 500;
-  autoSelectChrome: boolean;
-  TMP_PROFILE_DIR: string;
-  outFile?: number;
-  errFile?: number;
-  pidFile: string;
-  startingUrl: string;
-  chromeFlags: Array<string>;
-  chrome?: childProcess.ChildProcess;
+export interface LaunchedChrome {
+  pid: number;
   port: number;
+  kill: () => Promise<{}>;
+}
+
+export async function launch(opts: Options = {}): Promise<LaunchedChrome> {
+  opts.handleSIGINT = defaults(opts.handleSIGINT, true);
+
+  const instance = new Launcher(opts);
+
+  // Kill spawned Chrome process in case of ctrl-C.
+  if (opts.handleSIGINT) {
+    process.on(_SIGINT, async () => {
+      await instance.kill();
+      process.exit(_SIGINT_EXIT_CODE);
+    });
+  }
+
+  await instance.launch();
 
-  constructor(opts: {
-    startingUrl?: string,
-    chromeFlags?: Array<string>,
-    autoSelectChrome?: boolean,
-    port?: number
-  } = {}) {
+  return {pid: instance.pid!, port: instance.port, kill: async () => instance.kill()};
+}
+
+export class Launcher {
+  private tmpDirandPidFileReady = false;
+  private pollInterval: number = 500;
+  private pidFile: string;
+  private startingUrl: string;
+  private TMP_PROFILE_DIR: string;
+  private outFile?: number;
+  private errFile?: number;
+  private chromePath?: string;
+  private chromeFlags: string[];
+  private chrome?: childProcess.ChildProcess;
+  port: number;
+  pid?: number;
+
+  constructor(opts: Options = {}) {
     // choose the first one (default)
-    this.autoSelectChrome = defaults(opts.autoSelectChrome, true);
     this.startingUrl = defaults(opts.startingUrl, 'about:blank');
     this.chromeFlags = defaults(opts.chromeFlags, []);
-    this.port = defaults(opts.port, 9222);
+    this.port = defaults(opts.port, 0);
```

Comment:
it was getting a tad confusing with the multiple `this.port`'s... let's change this one to `this.requestedPort` ?